### PR TITLE
[layer context] Add interfaces for setBatch 

### PR DESCRIPTION
**Pull request by kparichay**
Add interfaces to layer context required by layer developer
when they override setBatch()

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM

File: nntrainer/tensor/weight.h [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Weight : public Var_Grad {
    *
    */
   typedef std::tuple<TensorDim, WeightInitializer, WeightRegularizer, float,
-                     bool, std::string>
+                     bool, const std::string>
```
Comment (jijoongmoon) [inline nntrainer/tensor/weight.h:59]: is it Dimension of Weight, WeightInitializer, Regularizer, regularizer constant and name? if it is, please add them into brief.

**Review by jijoongmoon — APPROVED** [review]
LGTM, except minor comments

File: nntrainer/tensor/var_grad.h [hunk 2]
```diff
@@ -30,7 +30,7 @@ class Var_Grad {
    * @brief Specification of the Var_Grad
    *
    */
-  typedef std::tuple<const TensorDim, bool, const std::string> Spec;
+  typedef std::tuple<TensorDim, bool, const std::string> Spec;
```
[the remark this answers is not in the document]
Comment (kparichay) [reply nntrainer/tensor/var_grad.h:35]: Yes, it is. Added in details.

File: nntrainer/tensor/weight.h (hunk 1 above)
Comment (kparichay) [reply nntrainer/tensor/weight.h:61]: Yes, it is. Added in details.

**Comment by kparichay** [comment]
Applied https://github.com/nnstreamer/nntrainer/pull/1312#discussion_r657037181 in this PR.

**Pull request merged**